### PR TITLE
[API-2108]UCD Documentation

### PR DESCRIPTION
**Pull request by emreyigit**
Documentation of python user code deployment and yaml job definition. 

Note: This docs will be valid after UCD is released in Viridian. Also the base image name is not defined yet officially. It should be fixed after it is created.

### CODE REVIEW

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:3]: basic job may have 2 or more stages.

Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:3]: typo: codding --> coding

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:5]:
```suggestion
NOTE: This feature is only available if your cluster is running on Viridian.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:234]:
```suggestion
TIP: This feature requires additional configuration on the map. You should enable _Event Journal_ for your map. See link:https://docs.hazelcast.com/hazelcast/latest/data-structures/event-journal#hide-nav[event journal documentation] for details.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:312]:
```suggestion
|No
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (ihsandemir) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:325]:
```suggestion
Assume that you have a map journal source, and the source contains `City` object. When a new object put the source map, it will be streamed to the pipeline. In transform step, the object type of `data` argument will be a key value pair since source is a map and holds key value pairs. The key value is an integer number in this example, and value is `City` object.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (emreyigit) [reply docs/modules/ROOT/pages/yaml-job-definition.adoc:312]: This is optinonal.

**Review by ihsandemir — APPROVED** [review]

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:429]:
```suggestion
|Defines the sink as JDBC
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:439]:
```suggestion
|Query statement to be run while inserting data to the JDBC. For further information on using JDBC as a sink, see the xref:integrate/jdbc-connector.adoc#jdbc-as-a-sink[] section of the JDBC Connector topic.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:462]:
```suggestion
|Defines the sink as Kafka.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:467]:
```suggestion
|Topic name to push.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:472]:
```suggestion
|The serializer used for the key.  
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:477]:
```suggestion
|The serializer used for the value.  
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:482]:
```suggestion
|Kafka properties to pass to the Kafka producer.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:502]:
```suggestion
This is the simplest sink option, where the data sinks to server logs.
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:539]:
```suggestion
|Name of the Map to use as sink.
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (yuce) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:1]:
```suggestion
= YAML Pipeline Definition
```

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (yuce) [inline docs/modules/ROOT/pages/yaml-job-definition.adoc:3]:
```suggestion
The stages of a Jet pipeline can be defined with in a yaml file without any Java code interaction. A basic streaming job consists of two or more stages. These are source, transform and sink stages. The transformation stage is optional, and a pipeline can have multiple transformation steps. In order to define these stages for your job, you can benefit from a yaml definition without any coding effort.
```

**Review by yuce — CHANGES_REQUESTED** [review]
Just two trivial ideas, will continue to review.

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (emreyigit) [reply docs/modules/ROOT/pages/yaml-job-definition.adoc]: Fixed at https://github.com/hazelcast/clc-docs/pull/10/commits/2e8f9c0178b3d0c6c738c19cf23c063f91c0d1d8

File: docs/modules/ROOT/pages/yaml-job-definition.adoc
Comment (emreyigit) [reply docs/modules/ROOT/pages/yaml-job-definition.adoc]: Fixed at https://github.com/hazelcast/clc-docs/pull/10/commits/2e8f9c0178b3d0c6c738c19cf23c063f91c0d1d8

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

**Review by ihsandemir — APPROVED** [review]

**Comment by yuce** [comment]
Let's please not merge this one until the following PR is reviewed and merged:
https://github.com/hazelcast/clc/pull/92

**Review by yuce — APPROVED** [review]

**Pull request merged**
